### ISSUE TITLE
Content Layer API inserts scripts from all collection entries into every generated page

### ISSUE THREAD

**Issue by jurajkapsz** [withastro/astro#12024]
### Astro Info

```block
Astro                    v4.15.6
Node                     v20.11.1
System                   Linux (x64)
Package Manager          pnpm
Output                   hybrid
Adapter                  @astrojs/node
Integrations             @astrojs/mdx
                         @astrojs/sitemap
```

### Describe the Bug

Switched to Content Layer API, having content collections with `mdx` files that include and call various [astro] components in their content.

```mdx
---
frontmatter...
---

import SomeComponent from "@components/some-component.astro";

Lorem ....

<SomeComponent/>
```

[astro] components can have scripts and styles.

When browsing website, I see scripts basically from all components included and executed on every page, even if they are not part of given page content.

### What's the expected result?

Only those assets should be part of the page that are included in content or template. This worked with `Content Collections`.

### Link to Minimal Reproducible Example

#

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by jurajkapsz** [comment]
Should I rather create an Astro 5 beta minimal reproduction, if any? Will be Content Layer API non-experimental also for Astro v4?

**Comment by ascorbic** [comment]
This should be fixed with https://github.com/withastro/astro/pull/12014. It will be in a release soon. Thanks for the report!

**Comment by jurajkapsz** [comment]
OK, thanks very much @ascorbic !

**Comment by jurajkapsz** [comment]
> This should be fixed with #12014. It will be in a release soon. Thanks for the report!

@ascorbic Any chance #12030 could make it into the Astro v4 as well? Trying to test fixed Content Layer API in Astro v4.15.8, but I'm bumping into this one.

**Comment by matthewp** [comment]
Closed by https://github.com/withastro/astro/pull/12014
